[1.12] DCOS-46088: get status for all deployments

natmegs opened the pull request:
Ensure status is retrieved from deployments, even when getStatus method is not available.

Closes DCOS-46088.

## Testing

Not sure if this will hold up as deployments change over time, but for me I tested using the 1.12 soak cluster.
- Go to Services
- Click on "X Deployments" in top right corner
- List of deployments in modal should match X

Compare this to same set of steps on soak cluster (not using this branch, just soak url). For me, it was 3 deployments but only 2 would show up in the modal (+ errors in dev console).

## Trade-offs

Code trade-offs, none that I can see. But it is weird that the format of the deployment items is not consistent. This issue came up because most of the items had all the regular "Service" properties, but there was one that only had 
```js
{
deployment: {...},
isStale: true,
serviceId: "/spark/spark-history-kerberos"
}
```
which is why the `getStatus` method was causing errors. Also id stored as property `serviceId` instead of `id` as expected. Is this typical of any "stale" service/application?

## Dependencies

None

## Screenshots

Before:
<img width="1043" alt="screen shot 2018-12-07 at 11 51 22 am" src="https://user-images.githubusercontent.com/19582796/49670404-7b074300-fa19-11e8-933e-398dbbe422cd.png">

After:
<img width="1043" alt="screen shot 2018-12-07 at 11 50 17 am" src="https://user-images.githubusercontent.com/19582796/49670422-8a868c00-fa19-11e8-92fd-b208d74acbe7.png">

natmegs commented:
@jaxesn Agree that it is strange. I'm wondering if this is typical for "stale" deployments/services? I'm not sure how to get a service into a stale state to replicate, and I can't find any other examples in other soak clusters

AmrAbdelrazik commented:
Can we reach out to the team who owns the API (likely the Marathon/Orchestration team) and confirm with them.

natmegs commented:
Ok, this makes a lot more sense now. Stale services do not have `getStatus` because of how we compile the data in `getChildTableData`, so I've adjusted accordingly and added a couple integration tests.

nLight commented:
@jaxesn you might want to reconsider your change request :D Given that CI is green

jaxesn commented:
Merge before ci gets you!
